app/js/FileHandler.js

```diff
@@ -175,3 +192,24 @@ async function _writeFileToDisk(bucket, key, opts) {
   )
   return LocalFileWriter.promises.writeStream(fileStream, key)
 }
+
+async function _wrapWithRetry(method, ...params) {
+  const maxTries = Settings.maxRetries || 5
+  let delay = Settings.retryDelay || 125
+  let tries = 0
+
+  while (true) {
+    try {
+      const result = await method(...params)
+      return result
+    } catch (err) {
+      if (OError.hasCauseInstanceOf(err, UpstreamError) && tries < maxTries) {
+        await sleep(delay)
```

Comment:
Let's add a random delay here. Otherwise we will hit the backend in large batches on second try and overload it again.